### DISCUSSION TITLE
How does accurancy depend on Index size

### DISCUSSION THREAD

**Discussion by ava57r** [qdrant/qdrant#2269] [General]
Hello

I wrote early this issue https://github.com/qdrant/qdrant/issues/2155

I tried to make index with 
`m = 64`
`efConstruct = 512`
and search 
`efSearch = 512`

I see that accuracy worst in a huge index than in a small index.
I see that you have tested the index with datasets up to 10 million.
1. How does the accuracy depend on the size?
2. Do `m` and `ef` have any limits?
3. Is the NHSW graph being constructed for each shard separately?
4. In the documentation, you recommend setting the number of shards from 2 to 4 per server, and what will be the overhead if I specify 8 shards per 1 server?

**Comment by timvisee** [comment]
1. This is not linear, I'm not sure what the function would be to express this. Generally speaking, the bigger the index, the more accurate it gets. A bigger index will take a significantly longer time to build though. You should probably do some testing with different parameters on your data set to see what settings suit you best with regards to accuracy and indexing time.
2. There are no hard limits. In practice it will be limited by the index time/cost and index traversal time.
3. Yes.

**Comment by ava57r** [comment]
2. some recommendations from other impl HNSW https://github.com/nmslib/hnswlib/blob/master/ALGO_PARAMS.md
